pkg/controllers/clusterresourceplacement/controller.go

```diff
@@ -1046,3 +1046,33 @@ func isRolloutCompleted(crp *fleetv1beta1.ClusterResourcePlacement) bool {
 func isCRPScheduled(crp *fleetv1beta1.ClusterResourcePlacement) bool {
 	return condition.IsConditionStatusTrue(crp.GetCondition(string(fleetv1beta1.ClusterResourcePlacementScheduledConditionType)), crp.Generation)
 }
+
+func emitPlacementStatusMetric(crp *fleetv1beta1.ClusterResourcePlacement) {
+	// Check CRP Scheduled condition.
+	status := "nil"
+	cond := crp.GetCondition(string(fleetv1beta1.ClusterResourcePlacementScheduledConditionType))
+	if !condition.IsConditionStatusTrue(cond, crp.Generation) {
+		if cond != nil {
+			status = string(cond.Status)
+		}
+		metrics.FleetPlacementStatusLastTimeStampSeconds.WithLabelValues(crp.Name, strconv.FormatInt(crp.Generation, 10), string(fleetv1beta1.ClusterResourcePlacementScheduledConditionType), status).SetToCurrentTime()
+		return
+	}
+
+	// Check CRP expected conditions.
+	expectedCondTypes := determineExpectedCRPAndResourcePlacementStatusCondType(crp)
+	for _, condType := range expectedCondTypes {
+		cond = crp.GetCondition(string(condType.ClusterResourcePlacementConditionType()))
+		if !condition.IsConditionStatusTrue(cond, crp.Generation) {
+			if cond != nil {
+				status = string(cond.Status)
+			}
+			metrics.FleetPlacementStatusLastTimeStampSeconds.WithLabelValues(crp.Name, strconv.FormatInt(crp.Generation, 10), string(condType.ClusterResourcePlacementConditionType()), status).SetToCurrentTime()
+			return
+		}
+	}
+
+	// Emit the "ClusterResourcePlacementCompleted" condition metric to indicate that the CRP has completed.
+	// This condition is used solely for metric reporting purposes.
+	metrics.FleetPlacementStatusLastTimeStampSeconds.WithLabelValues(crp.Name, strconv.FormatInt(crp.Generation, 10), "ClusterResourcePlacementCompleted", string(metav1.ConditionTrue)).SetToCurrentTime()
```

Comment:
Maybe just "Completed" is better? The other ones are named "Scheduled", "RolloutStarted", "Available", etc.